File: Java-Core/src/main/java/presto/declaration/SingletonCategoryDeclaration.java
package presto.declaration;

import presto.grammar.MethodDeclarationList;
import presto.grammar.Identifier;
import presto.utils.CodeWriter;
import presto.utils.IdentifierList;

public class SingletonCategoryDeclaration extends ConcreteCategoryDeclaration {

	public SingletonCategoryDeclaration(Identifier name, IdentifierList attributes, MethodDeclarationList methods) {
		super(name, attributes, null, methods);
	}
	
	@Override
	protected void categoryTypeToEDialect(CodeWriter writer) {
		writer.append("singleton");
	}

	@Override
	protected void categoryTypeToODialect(CodeWriter writer) {
		writer.append("singleton");
	}

	@Override
	protected void categoryTypeToPDialect(CodeWriter writer) {
		writer.append("singleton");
	}
}

File: Java-CodeServer/src/main/resources/js/prompto/declaration/SingletonCategoryDeclaration.js
var ConcreteCategoryDeclaration = require("./ConcreteCategoryDeclaration").ConcreteCategoryDeclaration;

function SingletonCategoryDeclaration(name, attributes, methods) {
    ConcreteCategoryDeclaration.call(this, name, attributes, null, methods);
    return this;
}

SingletonCategoryDeclaration.prototype = Object.create(ConcreteCategoryDeclaration.prototype);
SingletonCategoryDeclaration.prototype.constructor = SingletonCategoryDeclaration;

SingletonCategoryDeclaration.prototype.categoryTypeToEDialect = function(writer) {
    writer.append("singleton");
};

SingletonCategoryDeclaration.prototype.categoryTypeToODialect = function(writer) {
    writer.append("singleton");
};

SingletonCategoryDeclaration.prototype.categoryTypeToSDialect = function(writer) {
    writer.append("singleton");
};

exports.SingletonCategoryDeclaration = SingletonCategoryDeclaration;